nima/http2/webserver/src/main/java/io/helidon/nima/http2/webserver/spi/Http2SubProtocolProvider.java

```diff
@@ -16,26 +16,37 @@
 
 package io.helidon.nima.http2.webserver.spi;
 
-import io.helidon.config.Config;
+import io.helidon.nima.webserver.ProtocolConfigs;
+import io.helidon.nima.webserver.spi.ProtocolConfig;
 
 /**
  * {@link java.util.ServiceLoader} provider interface for HTTP/2 sub-protocols.
+ *
+ * @param <T> type of the protocol configuration used by the provider
  */
-public interface Http2SubProtocolProvider {
+public interface Http2SubProtocolProvider<T extends ProtocolConfig> {
+
+    /**
+     * Provider's type, also expected as the configuration node name.
+     *
+     * @return type of this provider, such as {@code grpc}
+     */
+    String protocolType();
 
     /**
-     * Provider's specific configuration node name.
+     * Type of supported configuration.
      *
-     * @return name of the node to request
+     * @return protocol config type
      */
-    String configKey();
+    Class<T> protocolConfigType();
 
     /**
      * Creates an instance of HTTP/2 sub-protocol selector.
      *
-     * @param config {@link io.helidon.config.Config} configuration node located on the node returned by {@link #configKey()}
+     * @param config configuration of this protocol
+     * @param configs configuration for possible nested protocols
      * @return new HTTP/2 sub-protocol selector
      */
-    Http2SubProtocolSelector create(Config config);
+    Http2SubProtocolSelector create(T config, ProtocolConfigs configs);
```

Comment:
here too - we should ideally have just a single config per

Reply:
This is intentional - protocols may have subprotocols that need to be configured. They also need to be configured just once (such as HTTP2 has to be configured just once, and it is used both to upgrade from HTTP/1 and as a connection selector).
So the first parameter is the configuraiton of THIS protocol, the `ProtocolConfigs` is a configuration of possible "sub-protocols"